rerun-failed-run/rerun-failed-run.yml

```diff
@@ -0,0 +1,73 @@
+name: Retrigger failed run action
+
+on:
+  workflow_dispatch:
+    inputs:
+      run_id:
+        description: The id of the run which we want to retrigger
+        required: true
+      error_message:
+        description: Custom error message to look for in the logs.
+        required: true
+      repository:
+        description: Github repository
+        required: true
+jobs:
+  rerun:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Import Secrets
+        id: secrets
+        uses: hashicorp/vault-action@v2.7.4
+        with:
+          url: ${{ secrets.VAULT_ADDR }}
+          method: approle
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+          secrets: |
+            secret/data/products/infra/ci/retrigger-gha-workflow RETRIGGER_APP_KEY;
+            secret/data/products/infra/ci/retrigger-gha-workflow RETRIGGER_APP_ID;
+
+      - name: Generate a GitHub token
+        id: github-token
+        uses: tibdex/github-app-token@v2
+        with:
+          app_id: ${{ steps.secrets.outputs.RETRIGGER_APP_ID }}
+          private_key: ${{ steps.secrets.outputs.RETRIGGER_APP_KEY }}
+
+      - name: Check previous run logs
+        id: check_logs
+        env:
+          GH_DEBUG: api
+        run: |
+          echo ${{ steps.github-token.outputs.token }} | gh auth login --with-token
+          echo "Fetching logs for run with id: ${{ inputs.run_id }} from ${{ inputs.repository }} repository."
+          # Fetch the logs from the previous run
+          logs=$(gh run view ${{ inputs.run_id }} -R ${{ inputs.repository }} --log)
+          echo "Logs coming from run with id: ${{ inputs.run_id }}"
+          echo "$logs"
+          echo "end logs"
+          # Check if the desired string is present in the logs
+          if echo "$logs" | grep -F "${{ inputs.error_message }}" >/dev/null; then
+            count=$(echo "$logs" | grep -F "${{ inputs.error_message }}" | wc -l)
+            echo "count: $count"
+            if [ "$count" -gt 2 ]; then
+                echo "::set-output name=rerun::true"
```

Comment:
`set-output` are deprecated directives and should not be used anymore: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

Reply:
I replaced it with  $GITHUB_OUTPUT